base/reflection.jl

```diff
@@ -470,31 +478,35 @@ function length(mt::MethodTable)
 end
 isempty(mt::MethodTable) = (mt.defs === nothing)
 
-uncompressed_ast(l::Method) = uncompressed_ast(l.lambda_template)
-uncompressed_ast(l::LambdaInfo) =
-    isa(l.code,Array{UInt8,1}) ? ccall(:jl_uncompress_ast, Array{Any,1}, (Any,Any), l, l.code) : l.code
+uncompressed_ast(m::Method) = uncompressed_ast(m, m.source)
+function uncompressed_ast(m::Method, s::CodeInfo)
+    if isa(s.code, Array{UInt8,1})
+        s = ccall(:jl_copy_code_info, Ref{CodeInfo}, (Any,), s)
+        s.code = ccall(:jl_uncompress_ast, Array{Any,1}, (Any, Any), m, s.code)
+    end
+    return s
```

Comment:
the callee doesn't own this return value. we would need to make a deepcopy if that's your concern, but this is not new.